Lecture_2/Exercise_3/mylist.c

```diff
@@ -0,0 +1,59 @@
+#include <stdlib.h>
+#include "mylist.h"
+
+TptrList ListAdd(TptrList *link, TptrList elem)
+{
+    if( elem != NULL ){
+	elem->ptrNext=*link;
+	*link=elem;
+    }
+    return elem;
+}
+
+void ListRemove(TptrList *link)
+{
```

Comment:
The function is not correct, pls review.

Reply:
Do you mean function ListRemove? As for me the everything is good. This function obtain pointer to link field that point to element for removing from list because this link field mast point to next to removed element after removing. Maybe you mean that it was needed to add a check for the passed argument if it is not NULL? Then such check should be added to the others functions.